client/wrapper_test.go

```diff
@@ -47,10 +47,10 @@ func TestWaitForHeight(t *testing.T) {
 
 			if tc.diff > 0 {
 				// if it is the future, make sure we get correct header
-				assert.Equal(t, desired, header.Height)
+				assert.Equal(t, true, desired >= header.Height)
 			} else {
 				// for the past, that we get the next header
-				assert.Equal(t, status.Height+1, header.Height)
+				assert.Equal(t, true, status.Height+1 >= header.Height)
```

Comment:
Oh... this is due to the fact that the header events and status.Height are not properly synchronized, right?

Some discrepancy there was causing failures in gaiad proxy from cosmos (they got status.Height, then called for validators at that height only to find that no such block existed... guess status.Height is not to be trusted. I will do a bigger fix later, but this is good for now)